Source/WebCore/css/ShorthandSerializer.cpp

```diff
@@ -185,19 +192,22 @@ bool ShorthandSerializer::subsequentLonghandsHaveInitialValues(unsigned startInd
     return true;
 }
 
-bool ShorthandSerializer::commonSerializationChecks(const ComputedStyleExtractor& properties)
+ShorthandSerializer::CommonSerializationChecksResult ShorthandSerializer::commonSerializationChecks(const ComputedStyleExtractor& properties)
 {
     ASSERT(length() && length() <= maxShorthandLength);
 
     ASSERT(m_shorthand.id() != CSSPropertyAll);
 
-    for (unsigned i = 0; i < length(); ++i)
-        m_longhandValues[i] = properties.propertyValue(longhandProperty(i));
-
-    return false;
+    for (unsigned i = 0; i < length(); ++i) {
+        auto longhandValue = properties.propertyValue(longhandProperty(i));
+        if (!longhandValue)
+            return CommonSerializationChecksResult::DidNotSetLonghandValues;
```

Comment:
This could have just returned `true` and the caller would have returned the null string from `m_result` or if I overlooked something and `m_result` is not guaranteed to be null this could set it to the null string. The enumeration and all the other change is unneeded.

The boolean doesn't mean "success", it means that the common checks have determined the result. That can and should include determining it's an error.

Reply:
Yeah that makes sense I went ahead and changed it to the new approach. I set `m_result` to the `emptyString()` since it looks like that is what was being returned previously.